OneSignalSDK/onesignal/core/src/main/java/com/onesignal/user/IUserStateObserver.kt

```diff
@@ -0,0 +1,15 @@
+package com.onesignal.user
```

Comment:
What do you think about putting the 3 new user state related files into a subpackage `package com.onesignal.user.state`? That way the high-level `com.onesignal.user` package just contains the interface still.